File: programmers/level-1/12943. 콜라츠 추측/solution.cc
#include <string>
#include <vector>

using namespace std;

int solution(int num) {
    int cnt = 0;
    long long n = num;
    while (n > 1) {
        if (cnt > 500) return -1;
        n = n % 2 ? (n * 3) + 1 : n / 2;
        cnt++;
    }
    return cnt;
}

File: programmers/level-1/12943. 콜라츠 추측/solution.js
/**
 * https://programmers.co.kr/learn/courses/30/lessons/12943
 * @param {number} n 
 */
function solution(n) {
    let cnt = 0;
    while (n > 1) {
        if (cnt > 500) return -1;
        n = n % 2 ? (n * 3) + 1 : n / 2;
        cnt += 1;
    }
    return cnt;
}